### PR TITLE
eks sample app

### PR DESCRIPTION
**Pull request by jhsinger-klotho**
eks sample app with tests

### CODE REVIEW

File: ts-eks/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+curl localhost:3000/users
+# > ["john"]%
+```
+
+## Compile and Deploy with Klotho
+
+run the terminal commands:
+```sh
+# Compile the app
+npx tsc && klotho . --app ts-microservices -p aws
```
Comment (ewucc) [inline ts-eks/README.md:34]: nit: might change the name to ts-eks for consistency

File: ts-eks/src/userInfo.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+const sequelize = setupSequelize();
+
+async function setupSequelize(): Promise<Sequelize> {
+
+  /** @klotho::persist  {
+   *   id = "sequelizeDB"
+   * }
+   * */ 
+  const sequelize = new Sequelize(`sqlite::memory:`, { logging: false });
```
Comment (ewucc) [inline ts-eks/src/userInfo.ts:24]: double checking if we need to have sequelize as part of this sample. I thought it was the ts-microservices sample initially. 

But yeh more so to take into consideration the spin up times and all that.

File: ts-eks/test/klotho.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+execution_units:
+    microsrv-api:
+        type: eks
+        pulumi_params:
+            nodeType: node
+    microsrv-users:
+        type: eks
+        pulumi_params:
+            nodeType: node
+    sequelize-main:
```
Comment (ewucc) [inline ts-eks/test/klotho.yaml:54]: fwiw this sequelize-main isn't present in the base klotho.yaml file (non test one)

Not sure if it's intended or not

Comment (ewucc) [reply ts-eks/test/klotho.yaml:54]: Also if it's meant to be fargate/node

File: ts-eks/test/pulumi/index.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import { PolicyPack, validateResourceOfType, StackValidationPolicy } from "@pulumi/policy";
+
+const lambdaPolicy: StackValidationPolicy = {
+    name: "lambda-test",
+    description: "Lambda integration tests.",
+    enforcementLevel: "mandatory",
+    validateStack: async (args, reportViolation) => {
+        const lambdas = args.resources.filter(r => r.isType(aws.lambda.Function));
```
Comment (ewucc) [inline ts-eks/test/pulumi/index.ts:10]: This is checking for lambdas which we don't have any, also I think the memory values are off

**Review by ewucc — CHANGES_REQUESTED** [review]

File: ts-eks/test/klotho.yaml (hunk 3 above)
Comment (jhsinger-klotho) [reply ts-eks/test/klotho.yaml:54]: the base klotho.yaml got pushed up by accident. want it to be fargate so we mix and match

File: ts-eks/src/userInfo.ts (hunk 2 above)
Comment (jhsinger-klotho) [reply ts-eks/src/userInfo.ts:24]: right, i just want to test the env vars for eks which has to be either RDS or elasticache, both take a while. I think its worth the tests even thought the timing stinks

File: ts-eks/test/pulumi/index.ts (hunk 4 above)
Comment (jhsinger-klotho) [reply ts-eks/test/pulumi/index.ts:10]: yeah im going to delete pulumi policies for now, just so we can get something in

File: ts-eks/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
+/**
+ * @klotho::execution_unit {
+ *   id = "microsrv-api"
```
Comment (yuval-klotho) [inline ts-eks/src/index.ts:3]: nit: possibly change this, too?

File: ts-eks/src/userInfo.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  await sequelize.sync({ alter: true });
+  console.log("connected");
+  return sequelize;
+}
+
+interface attribs
+  extends Model<InferAttributes<attribs>, InferCreationAttributes<attribs>> {
+    user: string,
+  age: string;
+  org: string;
```
Comment (yuval-klotho) [inline ts-eks/src/userInfo.ts:37]:
```suggestion
    age: string;
    org: string;
```

File: ts-eks/src/users.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+/**
+ * 
+ * @klotho::execution_unit {
+ *   id = "microsrv-users"
```
Comment (yuval-klotho) [inline ts-eks/src/users.ts:4]: nit: this name too

File: ts-eks/test/klotho.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
+app: ts-microservices
```
Comment (yuval-klotho) [inline ts-eks/test/klotho.yaml:1]: nit: this name too

File: ts-eks/test/klotho.yaml [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        redis:
+            type: elasticache
+            pulumi_params_by_type:
+                elasticache:
+                    nodeType: cache.t3.micro
+                    numCacheNodes: 1
+    pubsub:
+        type: sns
+execution_units:
+    microsrv-api:
```
Comment (yuval-klotho) [inline ts-eks/test/klotho.yaml:46]: if you do change the names as per the previous comments, here too

**Review by yuval-klotho — APPROVED** [review]
Some of the exec unit names could be updated imo, but I don't think that blocks this, which is on the critical path for today's work.

**Pull request merged**